megamek/src/megamek/client/bot/princess/FireControl.java

```diff
@@ -2527,6 +2527,7 @@ void loadAmmo(final Entity shooter,
             }
```

Comment:
Needs a `continue;` here to avoid an NPE:

https://lgtm.com/projects/g/MegaMek/megamek/snapshot/01ffc9b23acdb201d74d30b8fbf713894c95eb95/files/megamek/src/megamek/client/bot/princess/FireControl.java#x7c149b9bf1a4a99f:1

Reply:
Good catch on ArtilleryBayWeapon(77).  I don't think either line in Compute needs it - those seem to be 'fake' weapon attack actions created but not finalized to calculate damage.
I'll do some further playtesting with this on my next battle and see if I encounter unforeseen issues.